File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int[] w = new int[sizes.length];
        int[] h = new int[sizes.length];
        
        int idx = 0;
        for(int[] size: sizes){
            if(size[0] >= size[1]){
                w[idx] = size[0];
                h[idx] = size[1];
            }else{
                w[idx] = size[1];
                h[idx] = size[0];
            }

            idx++;
        }
        
        int answer = 
            (Arrays.stream(w).max().getAsInt()) *
            (Arrays.stream(h).max().getAsInt());
        
        return answer;
    }
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    newList = []
    for s in sizes:
        if s[0] < s[1]:
            newList.append([s[0],s[1]])
        else:
            newList.append([s[1],s[0]])
    maxWidth = 0
    maxHeight = 0
    for w,h in newList:
        if w > maxWidth:
            maxWidth=w
        if h > maxHeight:
            maxHeight=h
    
    return maxWidth * maxHeight